loki/transformations/argument_shape.py

```diff
@@ -80,6 +82,35 @@ def transform_subroutine(self, routine, **kwargs):  # pylint: disable=arguments-
                                          if d == ':']
                             vmap[arg] = arg.clone(type=arg.type.clone(shape=new_shape))
 
+                    elif re.search(r'\*$', str(arg.shape[-1])):
```

Comment:
Do we really need a regex here? Wouldn't the following work?
```suggestion
                    elif str(arg.shape[-1])[-1] == '*':
```